panic: runtime error: index out of range [-1]

Since https://github.com/nanovms/ops/commit/50a178d73871113ae0950f94e5d3ccc977fc22a8, I am getting:

```sh
# with any config.json, e.g.
$ cat ./config.json
{
  "Dirs" : ["./"]
}

$ ops run ./myApp -c ./config.json
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/nanovms/ops/fs.(*Manifest).AddFileTo(0xc000456b10, 0xc000456ae0, 0xc0000344b8, 0x1, 0x0, 0x0, 0xc0002744e0, 0x1)
        /home/am11/ops/fs/manifest.go:327 +0x7a0
github.com/nanovms/ops/fs.(*Manifest).AddFile(...)
        /home/am11/ops/fs/manifest.go:312
github.com/nanovms/ops/fs.(*Manifest).AddUserProgram(0xc000456b10, 0x0, 0x0)
        /home/am11/ops/fs/manifest.go:58 +0x1c3
github.com/nanovms/ops/lepton.BuildManifest(0xc0006c1650, 0x1acaa6f, 0xc00024a300, 0xc000210700)
        /home/am11/ops/lepton/image.go:309 +0x250
github.com/nanovms/ops/lepton.BuildImage(0x3b32ba8, 0x0, 0x0, 0x0, 0x0, 0xc0000c6ba0, 0x1f, 0x0, 0x0, 0x0, ...)
        /home/am11/ops/lepton/image.go:29 +0x2f
github.com/nanovms/ops/cmd.runCommandHandler(0xc00002a780, 0xc0004564b0, 0x1, 0x3)
        /home/am11/ops/cmd/cmd_run.go:64 +0x499
github.com/spf13/cobra.(*Command).execute(0xc00002a780, 0xc0004563f0, 0x3, 0x3, 0xc00002a780, 0xc0004563f0)
        /home/am11/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000248a00, 0x0, 0x96380, 0xc0000a4058)
        /home/am11/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /home/am11/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /home/am11/ops/ops.go:8 +0x27
```

it works fine until previous commit 285a23e8b2c093baec59cd3ac7f2ecb7aa6b3d9e.

With working commit, `fmt.Printf("%v\n", parts)` on line above (326) produces:

```
[lib x86_64-linux-gnu libnss_dns.so.2]
[etc ssl certs ca-certificates.crt]
[myApp]
[kernel]
[etc resolv.conf]
[proc sys kernel hostname]
[etc passwd]
[config.json]
[main.c]
[myApp]
[libc.so.6]
[ld-linux-x86-64.so.2]
```
with master, it is:

```
[lib x86_64-linux-gnu libnss_dns.so.2]
[etc ssl certs ca-certificates.crt]
[]
panic: runtime error: index out of range [-1]
```

--- a/cmd/cmd_image.go
+++ b/cmd/cmd_image.go
@@ -7,6 +7,7 @@ import (
 
 	"github.com/nanovms/ops/lepton"
 	api "github.com/nanovms/ops/lepton"
+	"github.com/nanovms/ops/types"
 	"github.com/spf13/cobra"
 )
 
@@ -275,9 +276,15 @@ func imageResizeCommandHandler(cmd *cobra.Command, args []string) {
 	config, _ := cmd.Flags().GetString("config")
 	config = strings.TrimSpace(config)
 
-	c := unWarpConfig(config)
+	c := &types.Config{}
+
+	err := unWarpConfig(config, c)
+	if err != nil {
+		exitWithError(err.Error())
+	}
+
 	globalFlags := NewGlobalCommandFlags(cmd.Flags())
-	err := globalFlags.MergeToConfig(c)
+	err = globalFlags.MergeToConfig(c)
 	if err != nil {
 		exitWithError(err.Error())
 	}
@@ -321,9 +328,14 @@ func imageSyncCommandHandler(cmd *cobra.Command, args []string) {
 	}
 
 	config, _ := cmd.Flags().GetString("config")
-	conf := unWarpConfig(config)
+	conf := &types.Config{}
+	err := unWarpConfig(config, conf)
+	if err != nil {
+		exitWithError(err.Error())
+	}
+
 	globalFlags := NewGlobalCommandFlags(cmd.Flags())
-	err := globalFlags.MergeToConfig(conf)
+	err = globalFlags.MergeToConfig(conf)
 	if err != nil {
 		exitWithError(err.Error())
 	}

--- a/cmd/flags_config.go
+++ b/cmd/flags_config.go
@@ -22,7 +22,11 @@ type ConfigCommandFlags struct {
 func (flags *ConfigCommandFlags) MergeToConfig(c *types.Config) (err error) {
 	if flags.Config != "" {
 
-		*c = *unWarpConfig(flags.Config)
+		if c == nil {
+			c = &types.Config{}
+		}
+
+		unWarpConfig(flags.Config, c)
 
 		return
 	}
@@ -35,8 +39,7 @@ func (flags *ConfigCommandFlags) MergeToConfig(c *types.Config) (err error) {
 }
 
 // unWarpConfig parses lepton config file from file
-func unWarpConfig(file string) *types.Config {
-	c := types.Config{}
+func unWarpConfig(file string, c *types.Config) (err error) {
 	data, err := ioutil.ReadFile(file)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "error reading config: %v\n", err)
@@ -44,16 +47,15 @@ func unWarpConfig(file string) *types.Config {
 	}
 	err = json.Unmarshal(data, &c)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "error config: %v\n", err)
-		os.Exit(1)
+		return fmt.Errorf("error config: %v\n", err)
 	}
 
 	c.VolumesDir = lepton.LocalVolumeDir
 	if c.Mounts != nil {
-		err = onprem.AddMountsFromConfig(&c)
+		err = onprem.AddMountsFromConfig(c)
 	}
 
-	return &c
+	return
 }
 
 // NewConfigCommandFlags returns an instance of ConfigCommandFlags

--- a/cmd/flags_config.go
+++ b/cmd/flags_config.go
@@ -47,7 +47,7 @@ func unWarpConfig(file string, c *types.Config) (err error) {
 	}
 	err = json.Unmarshal(data, &c)
 	if err != nil {
-		return fmt.Errorf("error config: %v\n", err)
+		return fmt.Errorf("error config: %v", err)
 	}
 
 	c.VolumesDir = lepton.LocalVolumeDir

--- a/cmd/flags_pkg.go
+++ b/cmd/flags_pkg.go
@@ -50,7 +50,11 @@ func (flags *PkgCommandFlags) MergeToConfig(c *types.Config) (err error) {
 		return errors.New("failed finding package manifest")
 	}
 
-	pkgConfig := unWarpConfig(manifestPath)
+	pkgConfig := &types.Config{}
+	err = unWarpConfig(manifestPath, pkgConfig)
+	if err != nil {
+		return err
+	}
 
 	c.Program = pkgConfig.Program
 	c.Version = pkgConfig.Version

--- a/lepton/image.go
+++ b/lepton/image.go
@@ -381,7 +381,7 @@ func createImageFile(c *types.Config, m *fs.Manifest) error {
 
 	err = mkfsCommand.Execute()
 	if err != nil {
-		return errors.Wrap(err, 1)
+		return err
 	}
 
 	return nil

